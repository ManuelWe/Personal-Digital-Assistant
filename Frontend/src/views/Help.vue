<template>
  <div
    id="help"
    class="hero-body"
  >
    <div class="container">
      <h1 class="title">
        Help Section
      </h1>
      <div class="columns">
        <div class="column">
          <h2 class="subtitle">
            What is Gunter?
          </h2>
          <p>
            Gunter is an intelligent Personal Digital Assistant.
            You can talk to him and he will talk to you.<br>
            Gunter offers four fundamental use cases which you can navigate to
            with the buttons in the Navbar or via speech.
          </p>
          <br>
          <h2 class="subtitle">
            The Use Cases
          </h2>
          <p>Commute: Notification when you have to get up</p>
          <p>Travel: Notification about low travel prices</p>
          <p>Restaurant: Shows you nice restaurants for your lunch break</p>
          <p>Trainer: Tells you when you can do sports</p>
        </div>
        <div class="column">
          <h2 class="subtitle">
            Everything you can say to Gunter
          </h2>
          <table class="table">
            <thead>
              <tr>
                <th>Command</th>
                <th>Result</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th>Commute</th>
                <th>Switch to commute use case</th>
              </tr>
              <tr>
                <th>Travel</th>
                <th>Switch to travel use case</th>
              </tr>
              <tr>
                <th>Restaurant</th>
                <th>Switch to restaurant use case</th>
              </tr>
              <tr>
                <th>Trainer</th>
                <th>Switch to trainer use case</th>
              </tr>
              <tr>
                <th>Preferences</th>
                <th>Switch to preferences</th>
              </tr>
              <tr>
                <th>Help</th>
                <th>Show this help page</th>
              </tr>
              <tr>
                <th>Time</th>
                <th>Gunter tells you the current time</th>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  created() {
    const utterance = new SpeechSynthesisUtterance('This is the help section. '
    + 'Here you can see an overview of what I can do!');
    utterance.rate = 1.4;
    utterance.lang = 'en-US';
    speechSynthesis.speak(utterance);
  },
};
</script>

<style scoped>
.subtitle {
  text-decoration: underline;
}
</style>
