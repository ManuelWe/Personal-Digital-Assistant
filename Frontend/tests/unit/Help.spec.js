import { createLocalVue, mount } from '@vue/test-utils';
import Buefy from 'buefy';
import Help from '@/views/Help.vue';
import persistentState from 'vue-persistent-state';
import SpeechService from '@/services/SpeechSynthesis';

const localVue = createLocalVue();
localVue.use(Buefy);
localVue.use(persistentState, {
  soundEnabled: true,
  microphoneEnabled: true,
  notificationsEnabled: false,
  notificationEndpoint: '',
});

jest.mock('@/services/SpeechSynthesis');

function factory() {
  return mount(Help, {
    localVue,
  });
}

describe('Help.vue', () => {
  describe('Elements are rendered correctly', () => {
    it('Component renders', () => {
      const wrapper = factory();
      expect(wrapper).toMatchSnapshot();
    });

    it('Title renders', () => {
      const wrapper = factory();
      expect(wrapper.find('.title').text()).toBe('Help Section');
    });
  });

  it('SpeecheService is called on page enter', () => {
    factory();
    expect(SpeechService.speak).toHaveBeenCalled();
  });
});
