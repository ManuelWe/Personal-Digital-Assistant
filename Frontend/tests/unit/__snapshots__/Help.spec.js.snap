// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help.vue Elements are rendered correctly Component renders 1`] = `
<div id="help" class="hero-body">
  <div class="container">
    <div class="columns">
      <div class="column">
        <h1 class="title">
          Help Section
        </h1> <br>
        <h2 class="subtitle">
          What is Gunter?
        </h2>
        <p>
          Gunter is an intelligent Personal Digital Assistant.
          You can talk to him and he will talk to you.<br>
          Gunter offers four fundamental use cases which you can navigate to
          with the buttons in the Navbar or via speech.
        </p> <br>
        <h2 class="subtitle">
          The Use Cases
        </h2>
        <p>morning-routine: Notification when you have to get up</p>
        <p>travel-planning: Notification about low travel prices</p>
        <p>lunch-break: Shows you nice restaurants for your lunch break</p>
        <p>personal-trainer: Tells you when you can do sports</p>
      </div>
      <div class="column">
        <h2 class="subtitle">
          Everything you can say to Gunter
        </h2>
        <table class="table">
          <thead>
            <tr>
              <th>Command</th>
              <th>Result</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th>morning-routine</th>
              <th>Switch to morning-routine use case</th>
            </tr>
            <tr>
              <th>travel-planning</th>
              <th>Switch to travel-planning use case</th>
            </tr>
            <tr>
              <th>lunch-break</th>
              <th>Switch to lunch-break use case</th>
            </tr>
            <tr>
              <th>personal-trainer</th>
              <th>Switch to personal-trainer use case</th>
            </tr>
            <tr>
              <th>Calendar</th>
              <th>Switch to calendar</th>
            </tr>
            <tr>
              <th>Preferences</th>
              <th>Switch to preferences</th>
            </tr>
            <tr>
              <th>Help</th>
              <th>Show this help page</th>
            </tr>
            <tr>
              <th>Time</th>
              <th>Gunter tells you the current time</th>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
